### PR TITLE
Delete SNI certificates via Poppy API [CDN-1176 and CDN-1175]

### PR DESCRIPTION
**Pull request by raman-balyan**

### CODE REVIEW

File: poppy/distributed_task/taskflow/task/delete_ssl_certificate_tasks.py [hunk 1]
```diff
@@ -29,8 +29,8 @@
 class DeleteProviderSSLCertificateTask(task.Task):
     default_provides = "responders"
 
-    def execute(self, providers_list, domain_name, cert_type,
-                project_id, flavor_id):
+    def execute(self, domain_name, cert_type, project_id,
+                providers_list=[], flavor_id=None):
```
Comment (merryldmello) [inline poppy/distributed_task/taskflow/task/delete_ssl_certificate_tasks.py]: providers_list cannot be an empty list as an optional value and flavor_id should always be present for the entire workflow to work.
http://docs.python-guide.org/en/latest/writing/gotchas/

File: poppy/transport/pecan/controllers/v1/ssl_certificates.py [hunk 2]
```diff
@@ -72,11 +72,21 @@ def post(self):
...
-        cert_type = pecan.request.GET.get('cert_type', 'san')
-
+        # Now, we are supporting 'san' and 'sni' cert types
         certificate_controller = \
             self._driver.manager.ssl_certificate_controller
+
+        try:
+            certs_info = certificate_controller.get_certs_info_by_domain(
+                domain_name=domain_name,
+                project_id=None)
```
Comment (merryldmello) [inline poppy/transport/pecan/controllers/v1/ssl_certificates.py]: project_id has been forced to a default of None, which would mean that the cert created, would have to be created with no project_id, which is not the case.

Comment (merryldmello) [reply poppy/transport/pecan/controllers/v1/ssl_certificates.py]: We could get the project_id from the headers information.

**Comment by cody-somerville** [comment]
@merryldmello is taking this over and splitting it into two separate PRs.

**Pull request closed without merging**